sdk/ai/azure-ai-resources/azure/ai/resources/operations/_azure_open_ai_deployment_operations.py

```diff
@@ -0,0 +1,80 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import uuid
+
+from typing import Iterable
+from azure.core.polling import LROPoller
+from azure.ai.ml import MLClient
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.resource.resources.models import DeploymentMode
+
+from .._restclient._azure_open_ai._client import MachineLearningServicesClient
+from .._utils._deployment_utils import get_empty_azure_open_ai_arm_template
+from ..entities import AzureOpenAIDeployment
+
+class AzureOpenAIDeploymentOperations():
+    def __init__(self, ml_client: MLClient, ai_client: MachineLearningServicesClient, **kwargs):
+        self._ml_client = ml_client
+        self._ai_client = ai_client
+        self._resource_management_client = ResourceManagementClient(self._ml_client._credential, self._ml_client.subscription_id)
+
+    def begin_create_or_update(self, deployment_name: str, deployment: AzureOpenAIDeployment) -> LROPoller[AzureOpenAIDeployment]:
```

Comment:
Ha, I run into this with indexes as well. I want `deployment_name` to be defined in and referenced from `deployment` object. But it seems not possible with tsp.

Reply:
yeah, it's a bit annoying :( but maybe it's better this way to match the autogen client as much as possible